### PR TITLE
[Backport-2.x][Tiered Caching] Indices Request cache stalekey management (#12625)

### PR DESCRIPTION
**Pull request by kiranprakash154**
### Description
Backports https://github.com/opensearch-project/OpenSearch/pull/12625

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kiranprakash154** [comment]
> ❌ Gradle check result for [4e937ca](https://github.com/opensearch-project/OpenSearch/commit/4e937cae2ff84d72c54b1f25089ec98477a215c4): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/35182/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md?rgh-link-date=2024-03-18T18%3A26%3A51Z#flaky-tests) unrelated to your change?

This was due to flaky test that has issue created - https://github.com/opensearch-project/OpenSearch/issues/10704

**Comment by kiranprakash154** [comment]
> ❌ Gradle check result for [4e937ca](https://github.com/opensearch-project/OpenSearch/commit/4e937cae2ff84d72c54b1f25089ec98477a215c4): [null](https://build.ci.opensearch.org/job/gradle-check/35202/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md?rgh-link-date=2024-03-18T21%3A24%3A19Z#flaky-tests) unrelated to your change?

This was due to flaky test that has issue created - https://github.com/opensearch-project/OpenSearch/issues/12236

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,7 +69,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix get task API does not refresh resource stats ([#11531](https://github.com/opensearch-project/OpenSearch/pull/11531))
 - Fix for deserilization bug in weighted round-robin metadata ([#11679](https://github.com/opensearch-project/OpenSearch/pull/11679))
 - onShardResult and onShardFailure are executed on one shard causes opensearch jvm crashed ([#12158](https://github.com/opensearch-project/OpenSearch/pull/12158))
-- Avoid overflow when sorting missing last on `epoch_millis` datetime field ([#12676](https://github.com/opensearch-project/OpenSearch/pull/12676))
```
Comment (andrross) [inline CHANGELOG.md:71]: Why did you remove this changelog entry?

**Review by andrross — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kiranprakash154) [reply CHANGELOG.md]: i didn't resolve conflict correctly, thanks for catching it. 
Pushed the change

**Review by andrross — APPROVED** [review]

**Review by sohami — APPROVED** [review]
